src/tools/build-manifest/src/main.rs

```diff
@@ -155,17 +155,19 @@ static TARGETS: &[&str] = &[
     "x86_64-unknown-hermit",
 ];
 
-static DOCS_TARGETS: &[&str] = &[
-    "aarch64-unknown-linux-gnu",
-    "i686-apple-darwin",
-    "i686-pc-windows-gnu",
-    "i686-pc-windows-msvc",
-    "i686-unknown-linux-gnu",
-    "x86_64-apple-darwin",
-    "x86_64-pc-windows-gnu",
-    "x86_64-pc-windows-msvc",
-    "x86_64-unknown-linux-gnu",
-    "x86_64-unknown-linux-musl",
+/// This allows the manifest to contain rust-docs for hosts that don't build
+/// docs.
+///
+/// Tuples of `(host_partial, host_instead)`. If the host does not have the
+/// rust-docs component available, then if the host name contains
+/// `host_partial`, it will use the docs from `host_instead` instead.
+///
+/// The order here matters, more specific entries should be first.
+static DOCS_FALLBACK: &[(&str, &str)] = &[
```

Comment:
Just to double check my understanding: this does not make us *lose* checking that e.g. x86_64-pc-windows-gnu shipped a rust-docs tarball, right? (If I'm reading the old code right, we never had checking of that kind).

It would generally be nice to assert somewhere that the builders we *want* to fallback *are* -- e.g., that aarch64-apple-darwin *isn't* producing a rust-docs tarball. I guess we can chalk that up to my (future) hope for a more general scheme that lets us check which tests/steps ran on which builders (or didn't run).

Reply:
Right, there weren't any checks before (except for the oddity that apple-darwin was hard-coded to require it).  I added an assert to check that the fallback exists.